wfe/context.go

```diff
@@ -65,6 +65,11 @@ func (th *topHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		logEvent.Endpoint = r.URL.String()
 	}
 	defer th.logEvent(logEvent)
+	defer func() {
+		if logEvent.Requester > 0 {
+			w.Header().Set("Boulder-Requester", fmt.Sprintf("%d", logEvent.Requester))
```

Comment:
I believe this will get called after `response.WriteHeader(code)`. We need to find another way to arrange for this to be called on every request that has a requester. One simple option that leads to a bit of duplication would be to set the header right after each `verifyPOST` call. This would be analagous to how we set the `logEvent.Requester` and `logEvent.Contacts` fields. Currently this is done inside `verifyPOST` (with one random exception in NewRegistration), but I would be okay with moving that to consistently happen at the call sites instead. This would be to avoid having to pass `response` into verifyPOST, which I would *really* not like to do.

Alternately we could refactor WFE so `verifyPOST` is called earlier in the handler stack, and the result passed along to lower values. This would give a natural single place to add the header, and would probably be a net improvement, but would be a very big refactoring.